crates/aggkit-prover/src/rpc.rs

```diff
@@ -180,24 +180,9 @@ impl AggchainProofGrpcService for GrpcService {
                     custom_chain_data: response.custom_chain_data.into(),
                 }))
             }
-            Err(e) => {
-                let e = match e.downcast::<aggchain_proof_service::Error>() {
-                    Ok(e) => e,
-                    Err(e) => return Err(Status::internal(e.to_string())),
-                };
-                match *e {
-                    aggchain_proof_service::Error::ProposerServiceError(
-                        proposer_service::Error::Client(
-                            proposer_client::Error::AggProofRequestFailed(
-                                jsonrpsee::core::client::error::Error::Call(e),
-                            ),
-                        ),
-                    ) if e.code() == -2000 => Err(Status::unavailable(
-                        "Proposer service has not built any proof yet",
-                    )),
-                    e => Err(Status::internal(e.to_string())),
-                }
-            }
+            // TODO: Return a different error when the proof is not yet ready.
+            // The gRPC API currently does not expose the status.
+            Err(e) => Err(Status::internal(e.to_string())),
```

Comment:
Wondering if we can recover the custom error here. Maybe we could even propagate the status, since we now have gRPC on both ends.

If you don't want to do it now, file an issue and I'll have a look as a follow up. @atanmarko

Reply:
Here it is https://github.com/agglayer/provers/issues/188